snapshotter/demux/snapshotter.go

```diff
@@ -158,6 +158,12 @@ func (s *Snapshotter) Commit(ctx context.Context, name string, key string, opts
 func (s *Snapshotter) Remove(ctx context.Context, key string) error {
```

Comment:
Would we implement other APIs that aren't in the Snapshotter interface? Treating `""` as a special "delete all" operation seems scary to be honest. If we would have different APIs in future, we should move these CleanupAll and DeteleAll there.

Reply:
So the `xxxAll` functions might be poorly named here. These are implemented on the `cache` and we just want to execute `Remove` on all cached snapshotter entries. In other words, broadcast the `Remove` or `Cleanup` request to all cached snapshotters. In theory a snapshot could be in use by multiple VMs and only available for free in a subset, but the remote snapshotter should be able to handle that case.

Reply:
Would `BroadcastRemoveRequest` be more clear than `RemoveAll`?

Reply:
RemoveAll is fine.

Reply:
If someone forget to set the namespace, we are not only deleting a VM's snapshot, but also all of VMs' snapshot. This seems dangerous to me.

That being said, I don't have better alternatives.

Reply:
The least I can do is to have a configuration option for enabling garbage collection via these broadcasts. So users would explicitly accept risk of enabling the feature and then forgetting to set the namespace would be user error. Also I have committed to some documentation for the demux snapshotter so that would outline the risk.